json_rpc/router.nim

```diff
@@ -226,35 +219,28 @@ macro rpc*(server: RpcRouter, path: string, body: untyped): untyped =
     setup = jsonToNim(parameters, paramsIdent)
     procBody = if body.kind == nnkStmtList: body else: body.body
 
-  if parameters.hasReturnType:
-    let ReturnType = parameters[0]
+  let ReturnType = if parameters.hasReturnType: parameters[0]
+                   else: ident "JsonNode"
+
+  # delegate async proc allows return and setting of result as native type
+  result.add quote do:
+    proc `doMain`(`paramsIdent`: JsonNode): Future[`ReturnType`] {.async.} =
+      `setup`
+      `procBody`
 
-    # delegate async proc allows return and setting of result as native type
+  if ReturnType == ident"JsonNode":
+    # `JsonNode` results don't need conversion
+    result.add quote do:
+      proc `procName`(`paramsIdent`: JsonNode): Future[StringOfJson] {.async, gcsafe.} =
+        return StringOfJson($(await `doMain`(`paramsIdent`)))
```

Comment:
Does this mean that we are no longer trapping the exceptions?

Reply:
No, I moved the exception handling in the code that is dispatching the request to the right user-supplied proc:
https://github.com/status-im/nim-json-rpc/pull/72/files#diff-380a803ec724e6ae58db2c9dcf966219R127-R135

This makes the overall code smaller and leads to less cyclomatic complexity in the invisible generated code (something I consider good for any macro-generated code in general).